src/esplog/mod.rs

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::io::*;
+use std::str::from_utf8;
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use byteorder::{LittleEndian, ReadBytesExt};
+
+use crate::esplog::mini_decompressor::{decompress_block, Quat, State};
+use crate::tags_impl::*;
+use crate::*;
+
+mod mini_decompressor;
+
+#[derive(Default)]
+pub struct EspLog {
+    pub model: Option<String>,
+}
+
+impl EspLog {
+    pub fn possible_extensions() -> Vec<&'static str> {
+        vec!["bin"]
+    }
+
+    pub fn detect<P: AsRef<std::path::Path>>(buffer: &[u8], _filepath: P) -> Option<Self> {
+        if buffer.len() > 7 && buffer[0..7] == [0x45, 0x73, 0x70, 0x4c, 0x6f, 0x67, 0x30] {

```

Comment:
`buffer[0..7] == b"EspLog0"` ill be equivalent, but more clear. the `b` prefix makes the char sequence &[u8] instead of &str